### ISSUE TITLE
$count with Singleton returning List

### ISSUE THREAD

**Issue by alvorteg** [OData/WebApi#2447]
Hi, I don't know if it is a bug, or missunderstanding. I think there is little documentation on singletons and their uses.

I want to have an endpoint to get a List of conflicts (strings), so I think the best approach is to model it as Singleton.

So I've modeled it as follows:
```
   edmBuilder.Singleton<List<string>>("Conflicts");
```

And I have a Controller:

```
  public class ConflictsController: ODataController
    {
        [EnableQuery]
        public IActionResult Get()
        {
            return Ok(new List<string>() { "id1", "id2", "id3" }.AsQueryable()); // This is only for testing.
        }
```

When I access to my URL: 

```
http://url/Conflicts
```

This returns a list of conflicts (strings) and this is working fine.

But now, in the Microsoft.AspNetCore.OData 7.5.5 (or above versions) if I want to do a ?$count=true to paginate those conflicts, I get the following error:

```
"Object of type 'System.Linq.EnumerableQuery`1[System.String]' cannot be converted to type 'System.Linq.IQueryable`1[System.Linq.IEnumerable`1[System.String]]'.",
```

Is this normal or is it a bug? How should it be fixed?

**_In the Microsoft.AspNetCore.OData 7.5.0 or 7.5.2 this worked fine_**... but i didn't find any related to this change on release notes.

Thanks

**Comment by xuzhg** [comment]
@habbes Would you please help take a look? Is there any recently commit related?

@alvorteg Basically, you should not define a singleton using 'dmBuilder.Singleton<IList<string>> ("Conflicts");`.
Is there any reason that you have to do like this?

**Comment by alvorteg** [comment]
@xuzhg Not really, I could create a "Conflicts" class with a property that is a list of strings (as you suggested in previous answers), but I wouldn't really need that intermediate class.

Anyway, I've tried that too and the $count=true still fails in 7.5.6.

Thanks!

**Comment by alvorteg** [comment]
@habbes @xuzhg Is there any news?

**Comment by mrink1074** [comment]
@alvorteg Just curious if you were able to resolve this issue?  I'm having the same issue when using $orderby.  My controller is named ComparablesController and it's returning IQueryable<ComparableViewModel> or IQueryable<Comparable> depending on Get method called, where these are custom object types.  The IQueryable<Comparable> works fine, but the IQueryable<ComparableViewModel> throws the same conversion error you were seeing.  Using Microsoft.AspNetCore.OData 8.0.10 nuget.  Thanks.
